### ISSUE TITLE
Suggest existing organisations / teams on signup with matching non-generic domains

### ISSUE THREAD

**Issue by kyle-ssg** [SolidStateGroup/bullet-train-api#3960]
### Is your feature request related to a problem? Please describe.

A common problem is that users incorrectly create a duplicate organisation because they haven't received / seen an invite from their team.

### Describe the solution you'd like.

- Allow users with non-generic emails to see existing organisations created by users with the same domain.
- Allow users to request an invite to an organisation

### Additional context

This was brought up as a way to improve time to first feature / improve onboarding

**Comment by dabeeeenster** [comment]
Hmm this would need to be designed very carefully so as not to leak information. I'm worried it would be hard to get any sort of canonical list of "non generic emails"

**Comment by dabeeeenster** [comment]
Maybe there is a way for us to not show the users but instead send an invite to the admin to request to join an org.

**Comment by kyle-ssg** [comment]
Yeah, I think we can get a good list of generic emails but it'll never be 100%. Maybe we should only suggest when the number of orgs is < than a suitable number. I think it'd just show the org name, not the users, or maybe mask some of the name.


It'd be worth looking at what other products do here, I've definitely seen this pattern.
